File: c/fizzbuzz.c
#include <stdio.h>

int main()
{
	int i = 0;
	for (i = 0; i < 100; ++i) {
		if (i % 3 == 0)
			printf("Fizz");
		if (i % 5 == 0)
			printf("Buzz");
		if (i % 3 != 0 && i % 5 != 0)
			printf("%d", i);
		printf("\n");
	}
	return 0;
}

File: scala/fizzbuzz.scala
#!/bin/sh
exec scala $0 $@
!#

def fizzbuzz(i: Int) =
  (i % 3, i % 5) match {
    case (0, 0) => "FizzBuzz"
    case (0, _) => "Fizz"
    case (_, 0) => "Buzz"
    case (_, _) => i.toString
  }

for (i <- 1 to 100) {
  println(fizzbuzz(i))
}

